test/orange/orangeinp/ConvexSurfaceBuilder.test.cc

```diff
@@ -0,0 +1,256 @@
+    }
+
+  protected:
+    Unit unit_;
+    UnitBuilder unit_builder_{&unit_, Tol::from_relative(1e-4)};
+    VariantTransform transform_;
+};
+
+void x_slab(ConvexSurfaceBuilder& build)
+{
+    build(Sense::outside, PlaneX{-0.5});
+    build(Sense::inside, PlaneX{0.5});
+}
+
+TEST_F(ConvexSurfaceBuilderTest, no_transform)
+{
+    {
+        SCOPED_TRACE("z_hemi");
+        auto css = this->make_state();
+        css.object_name = "zh";
+        ConvexSurfaceBuilder build{&unit_builder_, &css};
+        build(PlaneZ{0.0});
+        build(SphereCentered{1.0});
+        // clang-format off
+        static real_type const expected_local_bz[] = {-0.86602540378444,
+            -0.86602540378444, -0.86602540378444, 0.86602540378444,
+            0.86602540378444, 0, -1, -1, -1, 1, 1, 0, 1};
+        // clang-format on
+        EXPECT_VEC_SOFT_EQ(expected_local_bz, flattened(css.local_bzone));
+        EXPECT_VEC_SOFT_EQ(expected_local_bz, flattened(css.global_bzone));
+        static int const expected_nodes[] = {3, 5};
+        EXPECT_VEC_EQ(expected_nodes, to_vec_int(css.nodes));
+    }
+    {
+        SCOPED_TRACE("reverse_hemi");
+        auto css = this->make_state();
+        css.object_name = "rh";
+        ConvexSurfaceBuilder build{&unit_builder_, &css};
+        build(Sense::outside, PlaneZ{1e-5});
+        build(Sense::inside, SphereCentered{1.0});
+        // clang-format off
+        static real_type const expected_local_bz[] = {
+            -0.86602540378444, -0.86602540378444, 0,
+            0.86602540378444, 0.86602540378444, 0.86602540378444,
+            -1, -1, 0,
+            1, 1, 1,
+            1};
+        // clang-format on
+        EXPECT_VEC_SOFT_EQ(expected_local_bz, flattened(css.local_bzone));
+        EXPECT_VEC_SOFT_EQ(expected_local_bz, flattened(css.global_bzone));
+        static int const expected_nodes[] = {2, 5};
+        EXPECT_VEC_EQ(expected_nodes, to_vec_int(css.nodes));
+    }
+    {
+        SCOPED_TRACE("dedupe hemi");
+        auto css = this->make_state();
+        css.object_name = "dh";
+        ConvexSurfaceBuilder build{&unit_builder_, &css};
+        build(PlaneZ{1e-5});
+        build(SphereCentered{1.0});
+        // clang-format off
+        static real_type const expected_local_bz[] = {-0.86602540378444,
+            -0.86602540378444, -0.86602540378444, 0.86602540378444,
+            0.86602540378444, 0, -1, -1, -1, 1, 1, 0, 1};
+        // clang-format on
+        EXPECT_VEC_SOFT_EQ(expected_local_bz, flattened(css.local_bzone));
+        EXPECT_VEC_SOFT_EQ(expected_local_bz, flattened(css.global_bzone));
+
+        static int const expected_nodes[] = {3, 5};
+        EXPECT_VEC_EQ(expected_nodes, to_vec_int(css.nodes));
+    }
+    {
+        SCOPED_TRACE("slab");
+        auto css = this->make_state();
+        css.object_name = "sl";
+        ConvexSurfaceBuilder build{&unit_builder_, &css};
+        x_slab(build);
+        static real_type const expected_local_bz[] = {
```

Comment:
Point taken... I should update the "expected output" code. Historically this was because putting the array assignments next to each other would destroy clang-formatting assignment, but now we don't align on `=` so I can do that. I think this is an improvement in readability:
```c++
    {
        SCOPED_TRACE("slab");
        auto css = this->make_state();
        css.object_name = "sl";
        ConvexSurfaceBuilder build{&unit_builder_, &css};
        x_slab(build);

        static real_type const expected_local_bz[] = {
            -0.5, -inf, -inf, 0.5, inf, inf, -0.5, -inf, -inf, 0.5, inf, inf, 1};
        static real_type const expected_global_bz[] = {
            -0.5, -inf, -inf, 0.5, inf, inf, -0.5, -inf, -inf, 0.5, inf, inf, 1};
        static int const expected_nodes[] = {6, 8};
        EXPECT_VEC_SOFT_EQ(expected_local_bz, flattened(css.local_bzone));
        EXPECT_VEC_SOFT_EQ(expected_global_bz, flattened(css.global_bzone));
        EXPECT_VEC_EQ(expected_nodes, to_vec_int(css.nodes));
    }
```